ui/app/templates/components/secret-edit-display.hbs

```diff
@@ -9,6 +9,31 @@
     </div>
   {{/if}}
 
+  {{#if @showWriteWithoutReadWarning}}
+    {{#if (and @isV2 @model.failedServerRead)}}
+      <AlertBanner
+        @type="warning"
+        @message="We were not able to read metadata for this secret or the current version's data. Creating a new version of the secret with this form will not be able to use the check-and-set mechanism. If this is required on the secret, then you will need access to read the secret's metadata."
+        @class="is-marginless"
+        data-test-v2-no-cas-warning
+      />
+    {{else if @isV2}}
+      <AlertBanner
+        @type="warning"
+        @message="We were not able to read the current secret version to provide editing. Saving this form will create a new version of the secret and will utilize the available check-and-set mechanism."
+        @class="is-marginless"
+        data-test-v2-write-without-read
+      />
+    {{else}}
+      <AlertBanner
+        @type="warning"
+        @message="We were not able to read the current secret data. Saving using this form will overwrite the existing values."
```

Comment:
Same comment as above -- assuming that this warning message is only displayed when the users' policy prevents them from reading a secret, we should be explicit that it's the policy and not an unexpected server error.